### PR TITLE
Move @types dependencies to root package.json in devDependencies

### PR DESCRIPTION
**Pull request by idevelop**
All `@types` dependencies should be dev dependencies unless they're re-exported, which doesn't seem to be the case afaict. This ensures that production bundles don't include these unnecessary packages. 

In a very simple AWS Lambda that only imports `apollo-server-lambda` and `graphql` I reduced the size of the zip bundle from 6.2 MB to 5.2 MB just by explicitly excluding `node_modules/@types/**` in the `serverless.yml` file, which would have happened automatically if these packages were marked as dev dependencies.

I opened issue #4020 about this.

Based on the compile errors I was getting on my first pull request #4022, it seems that all type dependencies should be declared in the root project for compilation to work on CI. 

`tsc --build tsconfig.build.json` works fine on my machine.

### CODE REVIEW

**Comment by idevelop** [comment]
Not sure why the netlify preview failed, the error seems unrelated:

```
5:29:38 PM: Failed to launch chrome!
5:29:38 PM: Inconsistency detected by ld.so: dl-tls.c: 493: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
```

How do I trigger a rerun?

**Comment by idevelop** [comment]
Also, should I be manually bumping the package minor version for all the packages I'm affecting, or is this done automatically by some release script?

File: packages/apollo-server-lambda/package.json [hunk 1]
```diff
@@ -26,7 +26,6 @@
   },
   "dependencies": {
     "@apollographql/graphql-playground-html": "1.6.24",
-    "@types/aws-lambda": "^8.10.31",
```
Comment (abernix) [inline packages/apollo-server-lambda/package.json:29]: The `APIGatewayProxyEvent` type provided by this package is used within `apollo-server-express`:

https://github.com/apollographql/apollo-server/blob/84e1aa5ad3ee2acf410ca8984cfc68e90d4f8a66/packages/apollo-server-lambda/src/ApolloServer.ts#L1-L5


...and re-exported on the `CreateHandlerOptions` interface on the `onHealthCheck` property:

https://github.com/apollographql/apollo-server/blob/84e1aa5ad3ee2acf410ca8984cfc68e90d4f8a66/packages/apollo-server-lambda/src/ApolloServer.ts#L15-L25

File: packages/apollo-server-koa/package.json [hunk 2]
```diff
@@ -27,12 +27,6 @@
   "dependencies": {
     "@apollographql/graphql-playground-html": "1.6.24",
     "@koa/cors": "^2.2.1",
-    "@types/accepts": "^1.3.5",
-    "@types/cors": "^2.8.4",
-    "@types/koa": "^2.0.46",
-    "@types/koa-bodyparser": "^4.2.1",
-    "@types/koa-compose": "^3.2.2",
-    "@types/koa__cors": "^2.2.1",
```
Comment (abernix) [inline packages/apollo-server-koa/package.json:35]: At the very least, `koa` is used on the exported `KoaHandler` as `Koa.Context`:

https://github.com/apollographql/apollo-server/blob/aa3e23de9fbcde41a1bd3cca9c2eacf734bae582/packages/apollo-server-koa/src/koaApollo.ts#L1-L16

**Comment by abernix** [comment]
I've commented on https://github.com/apollographql/apollo-server/pull/4024 with some more perspective on this, but similarly, I'll close this for the reasons I noted in that Issue.

The lack of failures on the `tsc` compilation of server in your local checkout doesn't demonstrate that consuming applications wouldn't be affected, as they would be for the reason I noted in my PR review.

**Comment by abernix** [comment]
Sorry, meant to say I commented on https://github.com/apollographql/apollo-server/issues/4020.

**Pull request closed without merging**